### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by Tekhlay**
### Milestones 💪 

- [x] Write unit tests using Rspec.
- [x] Understand an existing codebase in a way that allows for adding tests.
- [x] Write Rspec unit tests with mocks.
- [x] Understand the TDD concept.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+      it 'should solve factorial for the given integers' do
+        expect(@solver.factorial(0)).to eq(1)
+        expect(@solver.factorial(1)).to eq(1)
+        expect(@solver.factorial(2)).to eq(2)
+        expect(@solver.factorial(3)).to eq(6)
+        expect(@solver.factorial(4)).to eq(24)
+        expect(@solver.factorial(5)).to eq(120)
+        expect(@solver.factorial(6)).to eq(720)
+      end
```
Comment (tresorsawasawa) [inline spec/solver_spec.rb:21]: - [ ] As required for the project, kindly ensure that The` #factorial` method has tests that handle the following case:
  * Negative value which raises an exception as an expectation

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Tekhlay and @developertariq 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 
-All tests are passing :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (rloterh) [reply spec/solver_spec.rb:21]: resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi  Tariq & Teklay,  

Your project is complete! 🏁  There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Good job with test cases!

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**